api/src/test/java/org/openmrs/module/appointments/service/impl/SMSServiceTest.java

```diff
@@ -0,0 +1,119 @@
+package org.openmrs.module.appointments.service.impl;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.openmrs.*;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.LocationService;
+import org.openmrs.api.context.Context;
+import org.openmrs.messagesource.MessageSourceService;
+import org.openmrs.module.appointments.connection.OpenmrsLogin;
+import org.openmrs.module.appointments.model.*;
+import org.openmrs.module.appointments.properties.AppointmentProperties;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.*;
+
+import static java.util.Arrays.asList;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.*;
+
+@PowerMockIgnore("javax.management.*")
+@PrepareForTest(Context.class)
+@RunWith(PowerMockRunner.class)
+public class SMSServiceTest {
+    public static final String PATIENT_IDENTIFIER = "GAN230901";
+    @Mock
+    private AdministrationService administrationService;
+
+    @Mock
+    private MessageSourceService messageSourceService;
+
+    @Mock
+    private LocationService locationService;
+
+    @Mock
+    private AppointmentVisitLocation appointmentVisitLocation;
+
+    @Mock
+    private Provider provider;
+
+    private GlobalProperty globalPropertyHelpdesk;
+    @Mock
+    private Patient patient;
+    @InjectMocks
+    private SMSService smsService;
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+        PowerMockito.mockStatic(Context.class);
+        when(Context.getAdministrationService()).thenReturn(administrationService);
+        when(Context.getMessageSourceService()).thenReturn(messageSourceService);
+        when(Context.getLocationService()).thenReturn(locationService);
+
+
+        when(administrationService.getGlobalProperty("sms.appointmentBookingSMSTemplate")).thenReturn("Appointment for {patientName}, {identifier} on {date} for {service} is booked at {facilityname}.");
```

Comment:
are these properties automatically created and defaulted to some value through config.xml?

Reply:
These properties are defined and created from clinic-config

Reply:
Added properties in cofig.xml too